ui/src/components/ConfigDetailsMenu/ConfigDetailsMenu.tsx

```diff
@@ -0,0 +1,142 @@
+  configName,
+}) => {
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+  const menuOpen = Boolean(anchorEl);
+  const [duplicateDialogOpen, setDuplicateDialogOpen] = useState(false);
+  const [deleteDialogOpen, setDeleteDialogOpen] = useState(false);
+  const [showDuplicateOption, setShowDuplicateOption] = useState<boolean>();
+
+  const { enqueueSnackbar } = useSnackbar();
+  const navigate = useNavigate();
+
+  const [fetchVersions] = useGetConfigurationVersionsLazyQuery({
+    onCompleted(data) {
+      const versionsData = new VersionsData(data);
+
+      if (versionsData.findCurrent() != null) {
+        setShowDuplicateOption(true);
+      }
+    },
+    variables: {
+      name: configName,
+    },
+  });
+
+  async function handleClick(e: React.MouseEvent<HTMLButtonElement>) {
+    setAnchorEl(e.currentTarget);
+    await fetchVersions();
+  }
+
+  function handleClose() {
+    setAnchorEl(null);
+  }
+
+  async function handleDelete() {
+    try {
+      await deleteResources([
+        {
+          kind: ResourceKind.CONFIGURATION,
+          metadata: {
+            name: configName,
+          },
+        },
+      ]);
+      enqueueSnackbar(`Deleted config ${configName}`, { variant: "success" });
+      setDeleteDialogOpen(false);
+      navigate("/configurations");
+    } catch (err) {
+      console.error(err);
+      enqueueSnackbar("Failed to delete config", { variant: "error" });
+    }
+  }
+
+  return (
+    <>
+      <IconButton
+        data-testid="config-menu-button"
+        id="config-menu-button"
+        aria-controls={menuOpen ? "config-menu" : undefined}
+        aria-haspopup="true"
+        aria-expanded={menuOpen ? "true" : undefined}
+        onClick={handleClick}
+      >
+        <MenuIcon />
+      </IconButton>
+      <Menu
+        id="config-menu"
+        anchorEl={anchorEl}
+        open={menuOpen}
+        onClose={handleClose}
+        MenuListProps={{
+          "aria-labelledby": "config-menu-button",
+        }}
+        anchorOrigin={{
+          vertical: "bottom",
+          horizontal: "left",
+        }}
+        transformOrigin={{
+          vertical: "top",
+          horizontal: "center",
+        }}
+      >
+        <MenuList>
+          {showDuplicateOption && (
+            <MenuItem onClick={() => setDuplicateDialogOpen(true)}>
+              <ListItemIcon>
+                <CopyIcon width="20px" />
+              </ListItemIcon>
+              <ListItemText>Duplicate current version</ListItemText>
+            </MenuItem>
+          )}
+          <MenuItem onClick={() => setDeleteDialogOpen(true)}>
+            <ListItemIcon>
+              <TrashIcon width="20px" stroke={colors.red[700]} />
+            </ListItemIcon>
+            <Typography color="error">Delete</Typography>
+          </MenuItem>
+        </MenuList>
+      </Menu>
+
+      <DuplicateConfigDialog
+        currentConfigName={asCurrentVersion(configName)}
+        open={duplicateDialogOpen}
```

Comment:
```suggestion
        open={duplicateDialogOpen}
        onClose={() => setDuplicateDialogOpen(false)}
```
I wasn't able to close the modal without this